### PR TITLE
add event-tracker parser for webmontag.de

### PR DESCRIPTION
**Pull request by amed**

### CODE REVIEW

File: data/webmondays_v.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+- name: 52. Webmontag
```
Comment (jhilden) [inline data/webmondays_v.yaml]: this should be deleted.

File: lib/event_tracker.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
+require './event_tracker_haystack.rb'
+
+divider
```
Comment (jhilden) [inline lib/event_tracker.rb]: naming suggstion `print_divider`.  this would be a little bit clearer on what it does

File: lib/event_tracker.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  puts "Extracting website content".green
+rescue
+  puts "Cannot connect to website".red
+  abort
+end
+
+# Create archive object that has the whole li of links.
+# archive => Class: Nokogiri::XML::NodeSet
+#http://www.rubydoc.info/github/sparklemotion/nokogiri/Nokogiri/XML/NodeSet
+archive = webmontag_wiki_site.css('.archive li div')
```
Comment (jhilden) [inline lib/event_tracker.rb]: maybe call it `archive_index`?

File: lib/event_tracker.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+if (archive.empty?)
+  puts "\".archive\" selector is empty from the website!".red
+  abort
+end
+
+puts "\n\t #{archive.length} records found".green
+
+
+# This array contains each event from wiki website {name, href}.
+archive_clean = Array.new
```
Comment (jhilden) [inline lib/event_tracker.rb]: why are you using a new array here?  have you considered just modifying the original archive array?

File: lib/event_tracker.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+archive.each_with_index do |item, index|
+    # extracting {name, href} form each link and put it in one array (array of hashes).
+    archive_clean.push({name: item.children[0].content.to_s, href: item.children[0].attributes['href'].to_s})
+end
+
+# Create Yaml event-blocks, each block is a hash.
+archive_blocked = Array.new
```
Comment (jhilden) [inline lib/event_tracker.rb]: as far as I understand this, the `archive_blocked` varialbe just holds a list of events, right?  so maybe just calling it `events` would be more appropiate (and calling the the class `Event`)?

File: lib/event_tracker.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+archive_clean.each_with_index do |item, index|
+    block = EventBlock.new(item, index)
+    archive_blocked.push(block.to_hash)
+end
+
+print "\n\n"
+
+puts "Creating Yaml file /data/webmondays.yaml ".green
```
Comment (jhilden) [inline lib/event_tracker.rb]: file extension shoud be `.yml` instead of `.yaml`

File: lib/event_tracker.rb [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+yaml_string = File.read "../data/webmondays.yaml"
+data = YAML.load yaml_string
+yaml_blob = archive_blocked.reverse + data
```
Comment (jhilden) [inline lib/event_tracker.rb]: what is this doing?  can't we simply dump/overwrite the .yml file with our parsed events?

File: lib/event_tracker.rb [hunk 8]
```diff
@@ -0,0 +1,55 @@
+require './event_tracker_haystack.rb'
```
Comment (jhilden) [inline lib/event_tracker.rb]: I don't understand the filename.  haystack??

File: lib/event_tracker_haystack.rb [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+    colorize(self, 33)
+  end
+
+  def blue
+    colorize(self, 34)
+  end
+end
+
+
+class EventBlock
```
Comment (jhilden) [inline lib/event_tracker_haystack.rb]: the `Event` class should be in its own file which is called like the class

File: lib/event_tracker_haystack.rb [hunk 10]
```diff
@@ -0,0 +1,99 @@
+require 'nokogiri'
+require 'open-uri'
+require 'yaml'
+
+WEBSITE = 'http://webmontag.de'
+FACEBOOK_URL = ''
+
+def divider
```
Comment (jhilden) [inline lib/event_tracker_haystack.rb]: this should go into some "utils" file

File: lib/event_tracker_haystack.rb [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+"wiki_url"
+"date"
+----------
+=end
+
+    @index
+    @event_element
+    @hash
+
+  def initialize (event_element, index)
```
Comment (jhilden) [inline lib/event_tracker_haystack.rb]: I would pass in the `href` and the `name` individually into the constructor.  the Event does not need to coupled to the notion of an element that holds the information.

Comment (jhilden) [reply lib/event_tracker_haystack.rb]: or actually I guess the `name` should be passed in as `date`

**Review by jhilden — CHANGES_REQUESTED** [review]

File: lib/event_tracker.rb (hunk 3 above)
Comment (jhilden) [reply lib/event_tracker.rb]: or maybe just "event_index"?

File: lib/event_tracker.rb (hunk 7 above)
Comment (amed) [reply lib/event_tracker.rb]: This was from early structure. We have changed the flow of the script.

File: lib/event_tracker_haystack.rb (hunk 11 above)
Comment (amed) [reply lib/event_tracker_haystack.rb]: I don't fully trust that name could be the link structure 100%

Comment (amed) [reply lib/event_tracker_haystack.rb]: I think it is okay to keep the (event_element, index) What is the point to make it individually passed?

File: lib/event_tracker.rb (hunk 4 above)
Comment (amed) [reply lib/event_tracker.rb]: "events_index" is not an array, it is Nokogiri::XML::NodeSet.
There is need to create this.

Comment (amed) [reply lib/event_tracker.rb]: events_index_array

**Review by jhilden — APPROVED** [review]

**Pull request merged**